File: CS/CS/WindowsApplication1/CustomEditor/ExtendedButtonEditViewInfo.cs
using System;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.XtraEditors.Controls;
using System.Collections.Generic;

namespace WindowsApplication1
{
    public class ExtendedButtonEditViewInfo : ButtonEditViewInfo
    {
        public ExtendedButtonEditViewInfo(RepositoryItem item)
            : base(item)
        {

        }
        protected override EditorButtonPainter CreateButtonPainter()
        {
            ExtendedSkinEditorButtonPainter painter = new ExtendedSkinEditorButtonPainter(LookAndFeel);
            painter.OwnerEditInfo = this;
            return painter;
        }

        // Fields...
        private Dictionary<EditorButton, string> _CustomButtonCaptions = new Dictionary<EditorButton,string>();

        public Dictionary<EditorButton, string> CustomButtonCaptions
        {
            get { return _CustomButtonCaptions; }
            set
            {
                _CustomButtonCaptions = value;
            }
        }
        public string GetButtonCaption(EditorButton button)
        {
            string text = string.Empty;
            if (!CustomButtonCaptions.TryGetValue(button, out text)) 
                text = button.Caption;
            return text;
        }

    }
}

File: VB/WindowsApplication1/CustomEditor/ExtendedButtonEditViewInfo.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.XtraEditors.Controls
Imports System.Collections.Generic

Namespace WindowsApplication1
	Public Class ExtendedButtonEditViewInfo
		Inherits ButtonEditViewInfo
		Public Sub New(ByVal item As RepositoryItem)
			MyBase.New(item)

		End Sub
		Protected Overrides Function CreateButtonPainter() As EditorButtonPainter
			Dim painter As New ExtendedSkinEditorButtonPainter(LookAndFeel)
			painter.OwnerEditInfo = Me
			Return painter
		End Function

		' Fields...
		Private _CustomButtonCaptions As New Dictionary(Of EditorButton, String)()

		Public Property CustomButtonCaptions() As Dictionary(Of EditorButton, String)
			Get
				Return _CustomButtonCaptions
			End Get
			Set(ByVal value As Dictionary(Of EditorButton, String))
				_CustomButtonCaptions = value
			End Set
		End Property
		Public Function GetButtonCaption(ByVal button As EditorButton) As String
			Dim text As String = String.Empty
			If (Not CustomButtonCaptions.TryGetValue(button, text)) Then
				text = button.Caption
			End If
			Return text
		End Function

	End Class
End Namespace